presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1690,16 +1726,59 @@ public void commit()
         metastore.commit();
     }
 
-    public static boolean isPartitionsSystemTable(SchemaTableName tableName)
+    public static Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
     {
-        return tableName.getTableName().endsWith(PARTITIONS_TABLE_SUFFIX) && tableName.getTableName().length() > PARTITIONS_TABLE_SUFFIX.length();
+        for (SystemTableHandler handler : ImmutableList.of(partitionsTableHandler, propertiesTableHandler)) {
```

Comment:
I think we can write this as a stream
```
return Stream.of(partitionsTableHandler, propertiesTableHandler)
        .filter(handler -> handler.matches(tableName))
        .map(handler -> handler.getSourceTableName(tableName))
        .findAny();

Reply:
that's my initial thought actually. But intellij suggests me to rewrite like this :) I will rewrite that back then.